registry/config.go

```diff
@@ -143,13 +149,24 @@ func (config *serviceConfig) LoadInsecureRegistries(registries []string) error {
 
 skip:
 	for _, r := range registries {
-		// validate insecure registry
 		if _, err := ValidateIndexName(r); err != nil {
 			// before returning err, roll back to original data
 			config.ServiceConfig.InsecureRegistryCIDRs = originalCIDRs
 			config.ServiceConfig.IndexConfigs = originalIndexInfos
 			return err
 		}
+		// validate insecure registry
+		if schemeRegex.MatchString(r) {
+			// Remove https?:// and warn
+			logrus.Warnf("insecure registry %s should not contain '%s://'", r, strings.Split(r, "://")[0])
+			r = schemeRegex.ReplaceAllString(r, "")
```

Comment:
It's totally a question of style, but this could be written without using regexp matching/substitution:

```go
if strings.HasPrefix(r, "http://") {
        logrus.Warnf("insecure registry %s should not contain 'http://'", r)
        r = strings.TrimPrefix(r, "http://")
} else if strings.HasPrefix(r, "https://")  {
        logrus.Warnf("insecure registry %s should not contain 'https://'", r)
        r = strings.TrimPrefix(r, "https://")
} else if validateNoScheme(r) != nil {
        ...
}
```

I find this a little easier to read, but it is more repetitive, so it's hard for me to justify the preference. Anyway, feel free to ignore.